src/classes/COFF_CannotDelete_TEST.cls

```diff
@@ -0,0 +1,202 @@
+        System.assertEquals(false, results[3].success);
+        System.assertEquals(Label.CannotDelete, results[3].errors[0].message);
+
+        //Verifies Course Offering with Course Offering Schedule was not deleted
+        System.assertEquals(false, results[5].success);
+        System.assertEquals(Label.CannotDelete, results[5].errors[0].message);
+
+        //Verifies Course Offering with Term Grade and Course Connection was not deleted
+        System.assertEquals(false, results[0].success);
+        System.assertEquals(Label.CannotDelete, results[0].errors[0].message);
+
+         //Verifies Course Offering with Term Grade and Course Offering Schedule was not deleted
+        System.assertEquals(false, results[2].success);
+        System.assertEquals(Label.CannotDelete, results[2].errors[0].message);
+
+        //Verifies Course Offering with Term Grade, Course Connection, and Course Offering Schedule was not deleted
+        System.assertEquals(false, results[1].success);
+        System.assertEquals(Label.CannotDelete, results[1].errors[0].message);
+    }
+
+    /*********************************************************************************************************
+    * @description Test method to test the following:
+    * (1) If Prevent_Course_Offering_Deletion__c is disabled in Hierarchy Settings, and a Course Offering
+    * has child records, it can be deleted.
+    * (2) If Prevent_Course_Offering_Deletion__c is disabled in Hierarchy Settings, and a Course Offering
+    * does not have any child records, it can be deleted.
+    */
+    @isTest
+    private static void testAllowCourseOfferingDeletion() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c
+                                                        (Account_Processor__c = UTIL_Describe_API.getHhAccRecTypeID(),
+                                                        Prevent_Course_Offering_Deletion__c = false));
+
+        Contact[] contacts = UTIL_UnitTestData_TEST.getMultipleTestContacts(6);
+        insert contacts;
+
+        List<Course_Offering__c> courseOfferings = new List<Course_Offering__c>();
+        Course_Offering__c courseOffering1 = UTIL_UnitTestData_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering2 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering3 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering4 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering5 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering6 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        Course_Offering__c courseOffering7 = UTIL_UnitTestdata_TEST.createCourseOffering(NULL, NULL);
+        courseOfferings.add(courseOffering1);
+        courseOfferings.add(courseOffering2);
```

Comment:
same here

Reply:
@skosgi1 , the reason why I'm adding them to the list "courseOfferings" is not for the insertion DML but because later on on line 213, I'm deleting all the Course Offerings. I didn't want to have to do multiple delete dmls for all 7 course offering records.

Reply:
my bad!